api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -24,18 +24,17 @@
 import org.apache.iceberg.expressions.Projections;
 
 /**
- * API for overwriting files in a table by filter expression.
+ * API for overwriting files in a table.
  * <p>
  * This API accumulates file additions and produces a new {@link Snapshot} of the table by replacing
- * all the files that match the filter expression with the set of additions. This operation is used
- * to implement idempotent writes that always replace a section of a table with new data.
+ * all the deleted files with the set of additions. This operation is used to implement idempotent
+ * writes that always replace a section of a table with new data or update/delete operations that
+ * eagerly overwrite files.
```

Comment:
I think this should note that the default validation mode is idempotent -- that is, the overwrite is correct and should be committed out regardless of other concurrent changes to the table.

It may also help to add examples. The normal idempotent write is "replace all the data for day D with query results", and the normal eager overwrite case is "replace these files with filtered versions and ensure no new data that would need to be filtered has been added".

Reply:
Added this below.

Reply:
Looks good.